frontend/model/contracts/chatroom.js

```diff
@@ -116,6 +117,23 @@ export function findMessageIdx (id: string, messages: Array<Object>): number {
   return -1
 }
 
+export function isAddedNewMessage (message: GIMessage): Object {
+  const { action, meta } = message.decryptedValue()
+  const rootState = sbp('state/vuex/state')
+  const me = rootState.loggedIn.username
+
+  if (action.endsWith('addMessage') ||
+    action.endsWith('join') ||
+    action.endsWith('rename') ||
+    action.endsWith('changeDescription') ||
+    action.endsWith('leave')) {
+    // we add new pending message in 'handleSendMessage' function in ChatMain.vue
+    return me !== meta.username
+  }
+
+  return false
+}
+
```

Comment:
Since this function isn't being used in this file, could you please move it into `ChatMain.vue` where it is used?

Also, this condition could be shortened and simplified, to something like:

```js
...
const me = rootState.loggedIn.username

return /.*(addMessage|join|rename|changeDescription|leave)$/.test(action) && me !== meta.username
```

(But verify that I got that regex correct...)

Reply:
Nice idea to use regular expression. And also agree to relocate the function.